DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -1,754 +1,866 @@
-                            Exception = new ArgumentParseException(converterContext.Parameter, null, "No value was provided for this parameter."),
-                            CommandObject = null
-                        });
-
-                        return null;
-                    }
-
-                    parsedArguments[parameter] = parameter.DefaultValue.Value;
-                }
-            }
-        }
-        catch (Exception error)
-        {
-            await this.extension.commandErrored.InvokeAsync(converterContext.Extension, new CommandErroredEventArgs()
-            {
-                Context = CreateCommandContext(converterContext, eventArgs, parsedArguments),
-                Exception = new ArgumentParseException(converterContext.Parameter, error),
-                CommandObject = null
-            });
-
-            return null;
-        }
-
-        return CreateCommandContext(converterContext, eventArgs, parsedArguments);
-    }
-}
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+using DSharpPlus.Commands.ArgumentModifiers;
+using DSharpPlus.Commands.ContextChecks;
+using DSharpPlus.Commands.EventArgs;
+using DSharpPlus.Commands.Exceptions;
+using DSharpPlus.Commands.Processors.SlashCommands.ArgumentModifiers;
+using DSharpPlus.Commands.Processors.SlashCommands.Localization;
+using DSharpPlus.Commands.Processors.SlashCommands.Metadata;
+using DSharpPlus.Commands.Trees;
+using DSharpPlus.Commands.Trees.Metadata;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace DSharpPlus.Commands.Processors.SlashCommands;
+
+public sealed class SlashCommandProcessor : BaseCommandProcessor<InteractionCreateEventArgs, ISlashArgumentConverter, InteractionConverterContext, SlashCommandContext>
+{
+    // Required for GuildDownloadCompleted event
+    public const DiscordIntents RequiredIntents = DiscordIntents.Guilds;
+
+    public IReadOnlyDictionary<Type, DiscordApplicationCommandOptionType> TypeMappings { get; private set; } = new Dictionary<Type, DiscordApplicationCommandOptionType>();
+    public IReadOnlyDictionary<ulong, Command> Commands => applicationCommandsMapping;
+
+    private static readonly List<DiscordApplicationCommand> applicationCommands = [];
+    private static IReadOnlyDictionary<ulong, Command> applicationCommandsMapping;
+
+    private readonly Regex nameLocalizationRegex = new(@"^[-_\p{L}\p{N}\p{IsDevanagari}\p{IsThai}]{1,32}$");
+    private readonly Regex descrtiptionLocalizationRegex = new("^.{1,100}$");
```

Comment:
should use the regex source generator

Reply:
Sealed class

Reply:
(I also attempted to cheese this by having a helper class, but the compiler didn't like it for reasons I still haven't figured out)

Reply:
worst comes to worst unseal it, but why can't you just use a file-private helper class that's not nested